### PR TITLE
feat: add fields via cli

### PR DESCRIPTION
**Pull request by shahzad31**
fixes https://github.com/elastic/synthetics/issues/962

example 

` npm run push --fields '{"env":"dev"}'`

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -233,6 +236,24 @@ export function getCommonCommandOpts() {
     '--match <name>',
     'run/push tests with a name or tags that matches a pattern'
   );
+  const fields = createOption(
+    '--fields <jsonstring>',
+    'add fields to the monitor(s) in the format key=value'
```
Comment (emilioalvap) [inline src/options.ts:241]:
```suggestion
    'add fields to the monitor(s) in the format { "key": "value"}
```

File: src/options.ts [hunk 2]
```diff
@@ -70,6 +70,9 @@ export async function normalizeOptions(
       : {};
 
   options.params = Object.freeze(merge(config.params, cliArgs.params || {}));
+  options.fields = Object.freeze(
+    merge(config.monitor.fields, cliArgs.fields || {})
```
Comment (emilioalvap) [inline src/options.ts:74]:
```suggestion
    merge(config.monitor?.fields, cliArgs?.fields || {})
```

**Review by emilioalvap — COMMENTED** [review]
@shahzad31 Added some tweaks to fix e2e pipeline, can we add a unit test to validate the propagation of filters options to each monitor?

**Review by vigneshshanmugam — COMMENTED** [review]
Looks good, Can you add a test for this option to `cli.test.ts`?

**Review by vigneshshanmugam — APPROVED** [review]

**Pull request merged**